vision/api/VisionTest/ProductSearchTest.cs

```diff
@@ -1,318 +1,365 @@
-        public void TestProductSearchGcs()
-        {
-            var output = _productSearch.Run("get_similar_products_gcs", _projectId, REGION_NAME, INDEXED_PRODUCT_SET, PRODUCT_CATEGORY, REF_IMAGE_GCS_URI, SEARCH_FILTER);
-            Assert.Equal(0, output.ExitCode);
-            Assert.Contains(INDEXED_PRODUCT_1, output.Stdout);
-        }
-
         [Fact]
         public void TestPurgeProductsInProductSet()
         {
-            var output = _productSearch.Run("create_product", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_DISPLAY_NAME, PRODUCT_CATEGORY);
+            var output = Run("create_product", PROJECT_ID, REGION_NAME, PRODUCT_ID, PRODUCT_DISPLAY_NAME, PRODUCT_CATEGORY);
             Assert.Equal(0, output.ExitCode);
 
-            _productSearch.Run("create_product_set", _projectId, REGION_NAME, PRODUCT_SET_ID, PRODUCT_SET_DISPLAY_NAME);
-            _productSearch.Run("add_product_to_set", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_SET_ID);
+            Run("create_product_set", PROJECT_ID, REGION_NAME, PRODUCT_SET_ID, PRODUCT_SET_DISPLAY_NAME);
+            Run("add_product_to_set", PROJECT_ID, REGION_NAME, PRODUCT_ID, PRODUCT_SET_ID);
 
-            output = _productSearch.Run("list_products", _projectId, REGION_NAME);
+            output = Run("list_products", PROJECT_ID, REGION_NAME);
             Assert.Contains(String.Format("Product id: {0}", PRODUCT_ID), output.Stdout);
 
-            _productSearch.Run("purge_products_in_product_set", _projectId, REGION_NAME, PRODUCT_SET_ID);
-            output = _productSearch.Run("list_products", _projectId, REGION_NAME);
+            Run("purge_products_in_product_set", PROJECT_ID, REGION_NAME, PRODUCT_SET_ID);
+            output = Run("list_products", PROJECT_ID, REGION_NAME);
             Assert.DoesNotContain(String.Format("Product id: {0}", PRODUCT_ID), output.Stdout);
         }
 
         [Fact]
         public void TestPurgeOrphanProducts()
         {
-            _productSearch.Run("create_product", _projectId, REGION_NAME, PRODUCT_ID, PRODUCT_DISPLAY_NAME, PRODUCT_CATEGORY);
-            var output = _productSearch.Run("list_products", _projectId, REGION_NAME);
+            Run("create_product", PROJECT_ID, REGION_NAME, PRODUCT_ID, PRODUCT_DISPLAY_NAME, PRODUCT_CATEGORY);
+            var output = Run("list_products", PROJECT_ID, REGION_NAME);
             Assert.Contains(String.Format("Product id: {0}", PRODUCT_ID), output.Stdout);
 
-            _productSearch.Run("purge_orphan_products", _projectId, REGION_NAME);
+            Run("purge_orphan_products", PROJECT_ID, REGION_NAME);
 
-            output = _productSearch.Run("list_products", _projectId, REGION_NAME);
+            output = Run("list_products", PROJECT_ID, REGION_NAME);
             Assert.DoesNotContain(String.Format("Product id: {0}", PRODUCT_ID), output.Stdout);
         }
+    }
 
-        public void Dispose()
+    // These tests all require products and indexes that live longer than the
```

Comment:
If I understand correctly what you are doing here, with XUnit this is usually done using [Class Fixtures](https://xunit.net/docs/shared-context#class-fixture).

Reply:
Actually, the resources created here need live beyond the whole test program.  It's a quirk of the API.  I think the indexes take a while to build.  So a class fixture would actually tear everything down too soon.